src/Workspaces/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,413 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.Json
+{
+    using static EmbeddedSyntaxHelpers;
+    using static JsonHelpers;
+
+    using JsonToken = EmbeddedSyntaxToken<JsonKind>;
+    using JsonTrivia = EmbeddedSyntaxTrivia<JsonKind>;
+
+    internal struct JsonLexer
+    {
+        public readonly ImmutableArray<VirtualChar> Text;
+        public int Position;
+
+        public JsonLexer(ImmutableArray<VirtualChar> text) : this()
+        {
+            Text = text;
+        }
+
+        public VirtualChar CurrentChar => Position < Text.Length ? Text[Position] : new VirtualChar((char)0, default);
+
+        public ImmutableArray<VirtualChar> GetSubPattern(int start, int end)
```

Comment:
`end` is generally `Position`. Consider adding an overload where end is implicit.

Reply:
done.  #resolved.